lib/matplotlib/animation.py

```diff
@@ -876,6 +883,180 @@ def _args(self):
                 + self.output_args)
 
 
+def _included_frames(frame_list, frame_format):
+    """frame_list should be a list of filenames"""
+    return INCLUDED_FRAMES.format(Nframes=len(frame_list),
+                                  frame_dir=os.path.dirname(frame_list[0]),
+                                  frame_format=frame_format)
+
+
+def _embedded_frames(frame_list, frame_format):
+    """frame_list should be a list of base64-encoded png files"""
+    template = '  frames[{0}] = "data:image/{1};base64,{2}"\n'
+    embedded = "\n"
+    for i, frame_data in enumerate(frame_list):
+        embedded += template.format(i, frame_format,
+                                    frame_data.replace('\n', '\\\n'))
+    return embedded
+
+
+# Taken directly from jakevdp's JSAnimation package at
+# http://github.com/jakevdp/JSAnimation
+@writers.register('html')
+class HTMLWriter(FileMovieWriter):
+    # we start the animation id count at a random number: this way, if two
+    # animations are meant to be included on one HTML page, there is a
+    # very small chance of conflict.
+    rng = random.Random()
+    supported_formats = ['png', 'jpeg', 'tiff', 'svg']
+    args_key = 'animation.html_args'
+
+    @classmethod
+    def isAvailable(cls):
+        return True
+
+    @classmethod
+    def new_id(cls):
+        #return '%16x' % cls.rng.getrandbits(64)
+        return ''.join(cls.rng.choice(string.ascii_uppercase)
+                       for x in range(16))
```

Comment:
Just use https://docs.python.org/3.6/library/uuid.html (whichever version you like) and drop rng?  (at least I don't think .rng should be exposed)

Reply:
Good idea, done.